addons/test_mail/tests/test_controller_attachment.py

```diff
@@ -0,0 +1,68 @@
+from itertools import chain, product
+
+import odoo
+from odoo.addons.mail.tests.common_controllers import MailControllerAttachmentCommon
+
+
+@odoo.tests.tagged("-at_install", "post_install", "mail_controller")
+class TestAttachmentController(MailControllerAttachmentCommon):
+    def test_independent_attachment_delete(self):
+        """Test access to delete an attachment"""
+        # Subtest format: (user, token, result)
+        denied_cases = product(
+            (self.guest, self.user_employee, self.user_portal, self.user_public),
+            (False, True),
+            (False,),
+        )
+        allowed_cases = product(self.user_admin, (False, True), (True,))
+        self._execute_subtests_delete(chain(denied_cases, allowed_cases))
```

Comment:
Would strongly advice finding a less not-readable way to write those tests ... as in the end they are both quite verbose, and impossible to understand when trying to understand what is tested. product, chain, ... look like smart stuff but in the end it obfuscates a lot the code.